### ISSUE TITLE
Proposed linter: space around operators

### ISSUE THREAD

**Issue by lencioni** [brigade/scss-lint#400]
It would be nice to have a linter that ensures there are spaces around operators. This could work similarly to ESLint's space-infix-ops rule: http://eslint.org/docs/rules/space-infix-ops.html

Example bad:

```scss
padding: $some-padding*2;
```

Example good:

```scss
padding: $some-padding * 2;
```

**Comment by nenadjelovac** [comment]
+1

**Comment by Anahkiasen** [comment]
:+1:

**Comment by mrpavlikov** [comment]
+1

**Comment by nenadjelovac** [comment]
+1

**Comment by aurfon** [comment]
+1

**Comment by srawlins** [comment]
I can take this one. There should be spaces around infix operators in:

* properties (eg `margin: 1px + 1px`)
* mixin inclusions (eg `@include my-mixin(1px + 1px)`)
* variable declarations (eg `$x = 1px + 1px`)
* control flow expressions (`@if`, `@for`, `@while`, `@each`) (eg `@if $x == 7`)

This linter should _not_ lint:

* non-evaluated operations, like `:nth-child(3n+1)` (selectors like this are the only cases I can think of)
* non-operations, like `font: 10px/8px` (see Sass's [Division and /](http://sass-lang.com/documentation/file.SASS_REFERENCE.html#division-and-slash) section)
* stringed operators, like `font: '1+1'`

This linter should also lint interpolated expressions (eg `content: "I ate #{5 + 10} pies!"`), but this might be really hard to do. I'll try.

I think there should be a configuration variable, `spaces`, which dictates the number of preferred spaces on each side of an operator, defaulting to 1. If this is overly complicated, it could instead accept `one_space` or `no_space`.

Did I miss anything?

**Comment by sds** [comment]
This looks like it would cover a majority of cases, @srawlins. Thanks for offering to take it on.

I don't think we need to support a configurable number of spaces (i.e. I think `space` or `no_space` is acceptable). I've never seen a codebase that used a number of spaces other than zero or one for _any_ language (but I'm happy to be shown an exception).

**Comment by sds** [comment]
Implemented in #509.
